thread-manager/src/config.rs

```diff
@@ -0,0 +1,70 @@
+use {
+    crate::{CoreAllocation, NativeConfig, RayonConfig, TokioConfig},
+    serde::{Deserialize, Serialize},
+    std::collections::HashMap,
+    toml::de::Error,
+};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(default)]
+pub struct ThreadManagerConfig {
+    /// Configurations for the native OS thread pools
+    pub native_configs: HashMap<String, NativeConfig>,
+    /// Mapping from lookup name to the config name for Native
+    pub native_runtime_mapping: HashMap<String, String>,
+
+    /// Configurations for the rayon runtimes
+    pub rayon_configs: HashMap<String, RayonConfig>,
+    /// Mapping from lookup name to the config name for Rayon
+    pub rayon_runtime_mapping: HashMap<String, String>,
+
+    /// Configurations for the Tokio runtimes
+    pub tokio_configs: HashMap<String, TokioConfig>,
+    /// Mapping from lookup name to the config name for Tokio
+    pub tokio_runtime_mapping: HashMap<String, String>,
+
+    pub default_core_allocation: CoreAllocation,
+}
+
+impl Default for ThreadManagerConfig {
+    // Default is mostly to keep Deserialize implementation happy. This is not intended to be a valid config
+    fn default() -> Self {
+        Self {
+            native_configs: HashMap::from([("default".to_owned(), NativeConfig::default())]),
+            native_runtime_mapping: HashMap::new(),
+            rayon_configs: HashMap::from([("default".to_owned(), RayonConfig::default())]),
+            rayon_runtime_mapping: HashMap::new(),
+            tokio_configs: HashMap::from([("default".to_owned(), TokioConfig::default())]),
+            tokio_runtime_mapping: HashMap::new(),
+            default_core_allocation: CoreAllocation::OsDefault,
+        }
+    }
+}
+
+impl ThreadManagerConfig {
+    /// Returns a valid "built-in" configuration for Agave that is safe to use in most cases.
+    /// Most operators will likely use this.
+    pub fn default_for_agave() -> Self {
```

Comment:
could we just make this the default?

Reply:
No as "default" config should be effectively empty (as it is used by Deserialize). Since this functionessentially parses Toml it would create an infinite loop.